pkg/ccl/changefeedccl/changefeedbase/settings.go

```diff
@@ -252,3 +252,13 @@ var EventConsumerWorkerQueueSize = settings.RegisterIntSetting(
 	int64(util.ConstantWithMetamorphicTestRange("changefeed.event_consumer_worker_queue_size", 16, 0, 16)),
 	settings.NonNegativeInt,
 ).WithPublic()
+
+// EventConsumerDispatcherBufferSize specifies the maximum number of events the dispatcher can buffer.
+var EventConsumerDispatcherBufferSize = settings.RegisterIntSetting(
+	settings.TenantWritable,
+	"changefeed.event_consumer_dispatcher_buffer_size",
+	"if changefeed.event_consumer_workers is enabled, this setting sets the maxmimum number of events"+
+		"which can be buffered by the dispatcher",
+	int64(util.ConstantWithMetamorphicTestRange("changefeed.event_consumer_dispatcher_buffer_size", 64, 0, 64)),
```

Comment:
I would like to see the same benchmark run with this setting set to 0.
The reason why I want to do this is that I want to test a hypothesis that with size of 0 performance should be almost identical to the old version.  Basically, this gives each worker a bit of buffer space so that they don't have to block on mutex  (to emit to sink) -- but with this set to 0, I suspect performance should be almost identical.

Reply:
Interestingly enough with it set to 0 it is _much_ slower, in my `order_line` test doing 235k messages per second compared to the 300k of master.  This is with the only difference being that we're sending it to the worker and the worker is picking it off.    Im surprised that just sending it over a channel to another worker compared to lock/unlock makes such a huge difference.